src/components/Search/Search.js

```diff
@@ -28,90 +45,92 @@ function Search() {
       <div className={classes.searchIcon}>
         <SearchIcon />
       </div>
-      <form className={classes.root} noValidate autoComplete="off" onSubmit={handleSearch}>
-        <FormattedMessage id="search">
-          {placeHolderText => (
-            <InputBase
-              value={searchQuery}
-              onChange={handleChange}
-              placeholder={`${placeHolderText}`}
-              classes={{
-                root: classes.inputRoot,
-                input: classes.inputInput,
-              }}
-              className="searchField"
-              inputProps={{ 'aria-label': 'search' }}
-            />
-          )}
-        </FormattedMessage>
-      </form>
-      <div
-        className={`position-absolute searchResultContainer ${
-          showResults && searchResults.length ? 'd-block' : ''
-        }`}
-      >
-        <ul className="px-2 py-2 m-0">
-          {searchResults.length && !isLoading ? (
-            <>
-              {searchResults.map(result => (
-                <Link
-                  className={classes.productLink}
-                  key={result.item.id}
-                  to={{
-                    pathname: `/product-detail/${result.item.id}`,
-                    state: { fromHomepage: true },
+
+      <ClickAwayListener onClickAway={handleClickAway}>
+        <div className={moreClasses.root}>
+          <form className={classes.root} noValidate autoComplete="off" onSubmit={handleSearch}>
+            <FormattedMessage id="search">
+              {placeHolderText => (
+                <InputBase
+                  onKeyUp={handleClick}
+                  onClick={handleClick}
+                  value={searchQuery}
+                  onChange={handleChange}
+                  placeholder={`${placeHolderText}`}
+                  classes={{
+                    root: classes.inputRoot,
+                    input: classes.inputInput,
                   }}
-                  onMouseDown={() =>
-                    handleRoute({
-                      pathname: `/product-detail/${result.item.id}`,
-                      state: { fromHomepage: true },
-                    })
-                  }
-                  tabIndex="0"
-                >
-                  {' '}
-                  <li className="py-3 px-3 searchedElement">
-                    <span className="d-flex pt-2 pr-2 pb-1 text-primary searchResultTitle">
+                  className="searchField"
+                  inputProps={{ 'aria-label': 'search' }}
+                />
+              )}
+            </FormattedMessage>
+          </form>
+          {isSearchContainerOpen ? (
+            <div className="position-absolute searchResultContainer">
+              <List className={classes.root}>
+                {searchResults.length ? (
+                  searchResults.map(result => (
+                    <ListItem
+                      button
+                      onClick={() =>
+                        handleRoute({
```

Comment:
the whole pathName is required? i mean can we manage pathName inside the function? and just pass the id?

Reply:
Yup, that's better

Reply:
Done.